src/app/project/board/address-template/address-template.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-address-template',
+  template: `
+      <div class="sub-details">
+          <h4>Address:</h4>
+          <address class="contents">
+              <p *ngIf="address['streetAddress']">{{ address['streetAddress'] }}</p>
+              <span *ngIf="address['postalCode']" style="margin-right: 6px;">{{ address['postalCode'] }}</span>
+              <span *ngIf="address['addressLocality']">{{ address['addressLocality'] }}</span>
+          </address>
+      </div>
+  `
+})
+export class AddressTemplateComponent {
+    @Input() address: object;
```

Comment:
you should define an interface and use as a type instead of `object`

e.g. 
```
export interface Address {
    streetAddress: string;
    postalCode: string;
    addressLocality: string;
}
```
so then you can use `address.streetAddress` in your template
if the parameter is optional, you can use ? (e.g. postalCode?: string;)

Reply:
I am defining interfaces in second draft of project metadata page. Please check [issue DSP-1183](https://dasch.myjetbrains.com/youtrack/issue/DSP-1183)